src/types_.py

```diff
@@ -168,41 +166,93 @@ def is_group(self) -> bool:
         """Whether the row is a group of pages."""
         return self.navlink.link is None
 
-    def to_markdown(self) -> str:
+    def is_external(self, server_hostname: str) -> bool:
+        """Whether the row is an external reference.
+
+        Args:
+            server_hostname: The hostname of the discourse server.
+
+        Returns:
+            Whether the item in the table is an external item.
+        """
+        if self.navlink.link is None:
+            return False
+        comparison_link = self.navlink.link.lower()
+        return comparison_link.startswith("http") and not comparison_link.startswith(
+            server_hostname.lower()
+        )
```

Comment:
Couldn't we only rely on `server_hostname` prefix ?
If `server_hostname` doesn't start with `http` it will be considered as external (so we probably need to ensure it starts with it)

Reply:
I'm wondering also if we shouldn't have some more validation on the URL, to avoid some risks, this link will endup on the public page of charmhub. That's probably not there that this validation will occur, but we want to avoid things like `http://test*some markdown or html*`

Reply:
Re validation, there is a further step to check that a HEAD request to the external link works in a future PR which I think will address that need, what do you think?

Reply:
The issue regarding just relying on `comparison_link` is that many of the tests don't use a discourse link, they just use `link 1` or something similar. If this line is changed, 50 tests fail. Ideally we would change this so that we get correct discourse links in the factories - the scope of this PR is quite large so I'll add a follow up issue if this is a reasonable path forward

Reply:
yes the `http` prefix should be enough - its just that some refactoring is needed to make that possible

Reply:
Yes, this may then simplify quite a bit the code, as also passing down the server_hostname would not be required. Also this seems also something is consistent with another line [here](https://github.com/canonical/upload-charm-docs/blob/4d1d52ba83db9239c24010f5f860699223087f88/src/index.py#L272)